src/main.py

```diff
@@ -72,3 +77,10 @@
     # get top k terms with the highest idf score (this is atm not very useful
     # but it will be used for the tf.idf similarity stuff
     top_keywords = get_top_idf_features(train_description_feature, 5)
+
+    # calculate similarity between train an testset job descriptions
+    # this is of high order complexity - test it on a subset of the data
+    corpus_list = pandas_vector_to_list(train_description_feature)
+    queries_list = pandas_vector_to_list(test_description_feature)
+    similarity = get_tfidf_similarity(corpus_list, queries_list)
+    relevant_documents = extract_relevant_documents(similarity)
```

Comment:
yes, this method is regularly used for clustering!
just feed the similarity matrix from (85) in your favourite clustering algorithm

although though clustering you loose information, and i dont quite understand how it would help us here (other than speedup maybe)